packages/app/src/cli/services/dev/processes/dev-session.ts

```diff
@@ -145,13 +142,16 @@ async function handleDevSessionResult(
       await printWarning(message.value, processOptions.stdout)
     }
   } else if (result.status === 'created') {
+    isDevSessionReady = true
     await printSuccess(`✅ Ready, watching for changes in your app `, processOptions.stdout)
   } else if (result.status === 'aborted') {
-    outputDebug('❌ Session update aborted (new change detected)', processOptions.stdout)
-  } else {
-    await printError(`❌ Error`, processOptions.stderr)
-    await printError(`└  ${result.error}`, processOptions.stderr)
+    outputDebug('❌ Session update aborted (new change detected or error in Session Update)', processOptions.stdout)
+  } else if (result.status === 'remote-error' || result.status === 'unknown-error') {
+    await processUserErrors(result.error, processOptions, processOptions.stdout)
   }
+
+  // If we failed to create a session, exit the process
+  if (!isDevSessionReady) process.exit(1)
```

Comment:
Should this not be an AbortError or such?

Reply:
Yeah, that makes more sense, changed